www/src/components/search-form.js

```diff
@@ -78,6 +77,7 @@ css.insert(`
   }
 
   .algolia-autocomplete .ds-dropdown-menu [class^="ds-dataset-"] {
+    max-height: calc(100vh - 7rem) !important;
```

Comment:
we're pretty guilty of this, but is `7rem` the header height? Is it possible to do something like this, just so it's obvious?

```js
const HEADER_HEIGHT = `7rem`

// later
max-height: calc(100vh - ${HEADER_HEIGHT}) !important;

```

Reply:
@DSchau Good call! We have constants for header and banner heights in https://github.com/gatsbyjs/gatsby/blob/57b57e1116157e66be66235ccf5e0004edda8302/www/src/utils/presets.js#L37-L38

![search dropdown](https://user-images.githubusercontent.com/21834/46877227-a538ec00-ce40-11e8-936a-4524da857b3e.png)

☝️ We position the dropdown slightly above the header – which I personally like and would keep - and want some whitespace at the bottom (which @Wattenberger's PR already takes care of), so maybe something along the lines of

```js
const dropdownOffsetTop = ...
const dropdownOffsetBottom = ... 

max-height: calc(100vh - ${presets.headerHeight} - ${presets.bannerHeight} - ${presets.dropdownOffsetTop} - ${dropdownOffsetBottom}) !important;
```

to get rid of the magic number for `max-height`? We could then also replace https://github.com/gatsbyjs/gatsby/blob/57b57e1116157e66be66235ccf5e0004edda8302/www/src/components/search-form.js#L21 with a `calc` that considers `dropdownOffsetTop`.

Reply:
@fk seems manageable to me! Thanks for the leg work digging into it!

Reply:
💯 @Wattenberger Let us know if you want to help us getting rid of those magic numbers! 😅

Reply:
I do! Down with magic numbers!
I added a constant right above the styles since the `searchDropdownOffsetTop` is scoped to that one component.
I also noticed that the search dialog was too high on the homepage and scooched that down.